sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/DynamicDestinations.java

```diff
@@ -164,10 +164,6 @@ void setSideInputAccessorFromProcessContext(DoFn<?, ?>.ProcessContext context) {
             DynamicDestinations.class,
             new TypeDescriptors.TypeVariableExtractor<
                 DynamicDestinations<T, DestinationT>, DestinationT>() {});
-    checkArgument(
-        descriptor != null,
-        "Unable to infer a coder for DestinationT, "
-            + "please specify it explicitly by overriding getDestinationCoder()");
```

Comment:
Catch the CannotProvideCoderException and throw a CannotProvideCoderException with the caught exception with a more meaningful error:
`Unable to infer a coder for DestinationT, please specify it explicitly by overriding getDestinationCoder()`